src/components/QuickOpenModal.js

```diff
@@ -204,6 +211,20 @@ export class QuickOpenModal extends Component<Props, State> {
     }
   };
 
+  gotoLocation = (location: ?GotoLocationType) => {
+    const { selectLocation, selectedSource } = this.props;
+    const selectedSourceId = selectedSource ? selectedSource.get("id") : "";
```

Comment:
This function isn't ever called in a situation where the selected source won't exist and neither will the `location.sourceId` so this shushes Flow.

The times when the selectedSource doesn't exist we're passing in the ID to use.